channeldb/graph.go

```diff
@@ -4090,51 +4165,60 @@ func fetchLightningNode(nodeBucket kvdb.RBucket,
 	return deserializeLightningNode(nodeReader)
 }
 
-func deserializeLightningNodeCacheable(r io.Reader, node *graphCacheNode) error {
+func deserializeLightningNodeCacheable(r io.Reader) (*graphCacheNode, error) {
 	// Always populate a feature vector, even if we don't have a node
 	// announcement and short circuit below.
-	node.features = lnwire.EmptyFeatureVector()
+	node := newGraphCacheNode(
+		route.Vertex{},
+		lnwire.EmptyFeatureVector(),
+	)
+
+	var nodeScratch [8]byte
 
 	// Skip ahead:
 	// - LastUpdate (8 bytes)
-	if _, err := r.Read(node.nodeScratch[:]); err != nil {
-		return err
+	if _, err := r.Read(nodeScratch[:]); err != nil {
```

Comment:
I digged into this a bit and I think many of the allocations might be just due to us using the `io.Reader` interface here, since both `Read` and `ReadFull` do allocate. We may in the future want to switch to just passing around the byte slices to these deserialize functions and just copy out the important parts.

Reply:
Seems related to: https://github.com/lightningnetwork/lnd/pull/4884